relay-server/src/metrics_extraction/sessions/types.rs

```diff
@@ -38,125 +179,138 @@ pub fn extract_session_metrics<T: SessionLike>(
         }
     };
 
-    let mut tags = BTreeMap::new();
-    tags.insert("release".to_owned(), attributes.release.clone());
-    if let Some(ref environment) = attributes.environment {
-        tags.insert("environment".to_owned(), environment.clone());
-    }
-    if let Some(client) = client {
-        tags.insert("sdk".to_owned(), client.to_owned());
-    }
+    let common_tags = CommonTags {
```

Comment:
adding this type instead of the references reintroduces that early clone you mentioned, although having owned types means you don't need to clone it the last time it's used here.